File: Fracto Project/Fracto.API/Models/Rating.cs
namespace Fracto.Models;
public class Rating
{
    public int RatingId { get; set; }
    public int DoctorId { get; set; }
    public int UserId { get; set; }
    public int Value { get; set; } // 1..5
    public Doctor? Doctor { get; set; }
    public User? User { get; set; }
}

File: Fracto Project/Fracto.frontend/src/app/services/rating.ts
// services/rating.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Rating, RatingCreateDto } from '../models/rating';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

@Injectable({ providedIn: 'root' })
export class RatingService {
  constructor(private http: HttpClient) {}

  rate(dto: RatingCreateDto) { return this.http.post(`${environment.apiUrl}/Rating`, dto); }
  byDoctor(doctorId: number): Observable<Rating[]> {
    return this.http.get<Rating[]>(`${environment.apiUrl}/Rating/doctor/${doctorId}`);
  }
}
